components/eamxx/src/physics/p3/tests/CMakeLists.txt

```diff
@@ -77,7 +77,11 @@ if (NOT SCREAM_BASELINES_ONLY)
 endif()
 
 if (SCREAM_ENABLE_BASELINE_TESTS)
-  set(BASELINE_FILE_ARG "-b ${SCREAM_TEST_DATA_DIR}/p3_run_and_cmp.baseline")
+  if (SCREAM_GENERATE_BASELINES)
+    set(BASELINE_FILE_ARG "-g -b ${SCREAM_BASELINES_DIR}/data/p3_run_and_cmp.baseline")
```

Comment:
Here, I'm assuming that if `SCREAM_GENERATE_BASELINES=ON`, we are not interested in running tests, but just in creating baselines. I think it's a fair assumption.